federation/pkg/federation-controller/replicaset/planner/planner.go

```diff
@@ -56,21 +58,27 @@ func NewPlanner(preferences *fed_api.FederatedReplicaSetPreferences) *Planner {
 // have all of the replicas assigned. In such case a cluster with higher weight has priority over
 // cluster with lower weight (or with lexicographically smaller name in case of draw).
 // It can also use the current replica count and estimated capacity to provide better planning and
-// adhere to rebalance policy.
+// adhere to rebalance policy. To avoid prioritization of clusters with smaller lexiconographical names
+// a semi-random string (like replica set name) can be provided.
 // Two maps are returned:
 // * a map that contains information how many replicas will be possible to run in a cluster.
 // * a map that contains information how many extra replicas would be nice to schedule in a cluster so,
 //   if by chance, they are scheudled we will be closer to the desired replicas layout.
 func (p *Planner) Plan(replicasToDistribute int64, availableClusters []string, currentReplicaCount map[string]int64,
-	estimatedCapacity map[string]int64) (map[string]int64, map[string]int64) {
+	estimatedCapacity map[string]int64, salt string) (map[string]int64, map[string]int64) {
```

Comment:
Thanks for educational link :)
Anyway - that suggests that it may not be the best name, because people won't know what it is (not everyone has a cryptography background).